lib/control.ml

```diff
@@ -128,13 +108,12 @@ implicit module Option = struct
     | Some a -> Some (f a)
 
   (* Applicative *)
-  let pure x = Some x
+  let return x = Some x
```

Comment:
Maybe it doesn't matter, but it seems strange to overload the "return" name for multiple modules, how would OCaml know if you do return some_applicative, to expect an applicative and not a monad?

Reply:
`return` must be the same as `pure` by definition, so that function doesn't need to be part of `Monad`. It's a historical accident that there are separate `pure` and `return` functions in Haskell - Applicative didn't used to exist, then it was squeezed in between Functor and Monad, so it needed a different name to `return`. But this suggests (and I agree) that `return` is a better basic name for the concept than `pure`.

Reply:
I was thinking `return` was "nicer", and it does make sense in `do` notation, but upon reflection `pure` is a bit clearer. But yes, this is bikeshedding - I'll just keep it as is to avoid churn.